test/unittests/test_excel2json/test_utils.py

```diff
@@ -179,6 +179,33 @@ def test_get_comments(self) -> None:
         returned_none = utl.get_comments(original_df.loc[1, :])
         self.assertIsNone(cast(None, returned_none))
 
+    def test_add_optional_columns(self) -> None:
+        original_df = pd.DataFrame(
+            {
+                "comment_en": ["text_en", pd.NA],
+                "comment_it": ["text_it", pd.NA],
+                "comment_rm": [pd.NA, pd.NA],
+            }
+        )
+        optional_cols = {"comment_en", "comment_de", "comment_fr", "comment_it", "comment_rm"}
+        expected_df = pd.DataFrame(
+            {
+                "comment_de": [pd.NA, pd.NA],
+                "comment_en": ["text_en", pd.NA],
+                "comment_fr": [pd.NA, pd.NA],
+                "comment_it": ["text_it", pd.NA],
+                "comment_rm": [pd.NA, pd.NA],
+            }
+        )
+        returned_df = utl.add_optional_columns(df=original_df, optional_col_set=optional_cols)
+        # as the columns are extracted via a set, they are not sorted and may appear in any order,
+        # this would cause the validation to fail
+        returned_df = returned_df.sort_index(axis=1)
+        assert_frame_equal(expected_df, returned_df)
+        # here all the columns exist
+        returned_df = utl.add_optional_columns(df=expected_df, optional_col_set=optional_cols)
+        assert_frame_equal(expected_df, returned_df)
```

Comment:
```suggestion

        # if all columns exist, the df should be returned unchanged
        unchanged_df = utl.add_optional_columns(df=expected_df, optional_col_set=optional_cols)
        assert_frame_equal(expected_df, unchanged_df)
```

... just to make it more clear